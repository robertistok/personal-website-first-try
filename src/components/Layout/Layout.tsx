import React from "react";
import Helmet from "react-helmet";
import styles from "./Layout.module.scss";

interface LayoutProps {
  children: React.ReactNode;
  title: string;
  description?: string;
}

const Layout: React.FunctionComponent<LayoutProps> = ({
  children,
  title,
  description
}: LayoutProps): React.ReactElement => (
  <div className={styles["layout"]}>
    <Helmet>
      <html lang="en" />
      <title>{title}</title>
      <meta name="description" content={description} />
      <meta property="og:site_name" content={title} />
      <meta name="twitter:card" content="summary" />
      <meta name="twitter:title" content={title} />
    </Helmet>
    {children}
  </div>
);

export default Layout;
