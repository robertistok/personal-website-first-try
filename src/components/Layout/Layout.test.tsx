import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Layout from "./Layout";

describe("<Layout />", () => {
  const props = {
    children: "test",
    description: "test",
    title: "test"
  };

  it("renders", () => {
    shallow(<Layout {...props} />);
  });

  it("renders and matches snapshot", () => {
    const wrapper = shallow(<Layout {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
