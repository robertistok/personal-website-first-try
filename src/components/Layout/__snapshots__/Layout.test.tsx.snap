// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders and matches snapshot 1`] = `
<div
  className="layout"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <html
      lang="en"
    />
    <title>
      test
    </title>
    <meta
      content="test"
      name="description"
    />
    <meta
      content="test"
      property="og:site_name"
    />
    <meta
      content="summary"
      name="twitter:card"
    />
    <meta
      content="test"
      name="twitter:title"
    />
  </HelmetWrapper>
  test
</div>
`;
